internal/commands/root.go

```diff
@@ -37,12 +36,19 @@ func NewRootCmd(use string, dockerCli command.Cli) *cobra.Command {
 				return cmd.GenBashCompletion(dockerCli.Out())
 			case "zsh":
 				return cmd.GenZshCompletion(dockerCli.Out())
-			case "":
-				// Actually unset
-				return nil
 			default:
-				return fmt.Errorf("%q is not a supported shell", completion)
+				if len(completion) > 0 {
```

Comment:
nit: We tend to use `if completion != ""` as a way to check if a string is not empty.